modules/packages/Python.chpl

```diff
@@ -2816,6 +2876,157 @@ module Python {
                      interpreter.toPythonInner(item));
       this.interpreter.checkException();
     }
+
+    /*
+      Set an item in the list. Equivalent to calling
+      ``obj[bounds.low:bounds.high+1] = items`` in Python.
+
+      Note that providing more or less items than specified by `bounds` will
+      change the length of the list.
+
+      :arg bounds: The indices of the items to set.
+      :arg item: The item to set.
+    */
+    proc set(bounds: range(?), items: ?) throws {
+      if (bounds.strides != strideKind.one) {
+        compilerError("cannot call `set()` on a Python list with a range with stride other than 1");
+      }
+
+      var ctx = chpl_pythonContext.enter();
+      defer ctx.exit();
+
+      if (bounds.hasLowBound() && bounds.hasHighBound()) {
+        PyList_SetSlice(this.getPyObject(), bounds.low.safeCast(Py_ssize_t),
+                        bounds.high.safeCast(Py_ssize_t) + 1,
+                        interpreter.toPythonInner(items));
+
+      } else if (!bounds.hasLowBound() && bounds.hasHighBound()) {
+        PyList_SetSlice(this.getPyObject(), min(Py_ssize_t),
+                        bounds.high.safeCast(Py_ssize_t) + 1,
+                        interpreter.toPythonInner(items));
+
+      } else if (bounds.hasLowBound() && !bounds.hasHighBound()) {
+        PyList_SetSlice(this.getPyObject(), bounds.low.safeCast(Py_ssize_t),
+                        max(Py_ssize_t), interpreter.toPythonInner(items));
+
+      } else {
+        PyList_SetSlice(this.getPyObject(), min(Py_ssize_t),
+                        max(Py_ssize_t), interpreter.toPythonInner(items));
+      }
+      this.interpreter.checkException();
+    }
+
+    /*
+      Insert an item into the list at the specified index.  Equivalent to
+      calling ``obj.insert(index, item)`` in Python.
+
+      :arg idx: The index to insert the item at
+      :arg item: The item to append
+    */
+    proc insert(idx: int, item: ?) throws {
+      var ctx = chpl_pythonContext.enter();
+      defer ctx.exit();
+
+      PyList_Insert(this.getPyObject(), idx.safeCast(Py_ssize_t),
+                    interpreter.toPythonInner(item));
+      this.interpreter.checkException();
+    }
+
+    /*
+      Append an item to the end of the list.  Equivalent to calling
+      ``obj.append(item)`` in Python.
+
+      :arg item: The item to append
+    */
+    proc append(item: ?) throws {
+      var ctx = chpl_pythonContext.enter();
+      defer ctx.exit();
+
+      PyList_Append(this.getPyObject(), interpreter.toPythonInner(item));
+      this.interpreter.checkException();
+    }
+
+    /*
+      Extend the list with the contents of `iterable`.  Equivalent to calling
+      ``obj.extend(iterable)`` in Python.
+
+      .. warning::
+
+         This method is only support for Python 3.13 or later!
+
+      :arg iterable: something that can be iterated over to extend the list
+    */
```

Comment:
Maybe a nit, but a `.. note` seems more appropriate here than a `.. warning`. And maybe a description of what happens if you don't call it with Python 3.13+